src/tool_cb_wrt.c

```diff
@@ -234,7 +234,8 @@ size_t tool_write_cb(char *buffer, size_t sz, size_t nmemb, void *userdata)
 #ifdef WIN32
   fhnd = _get_osfhandle(fileno(outs->stream));
   if(isatty(fileno(outs->stream)) &&
-     GetConsoleScreenBufferInfo((HANDLE)fhnd, &console_info)) {
+     GetConsoleScreenBufferInfo((HANDLE)fhnd, &console_info) &&
+     GetACP() != CP_UTF8) {
```

Comment:
This should probably use `GetConsoleOutputCP` instead.
I recall something about console corrupting UTF8 output though when a multibyte character lands on the end of the buffer, not sure if this is a good idea then.

Reply:
Good point, and quite possible. The goal would be to avoid the CRT doing any translations and just passing it to `WriteConsole()` internally. Reading the UCRT [source code](https://github.com/huangqinjin/ucrt), and still cannot decide which would be the one we need. It seems we'd need the handle opened in text mode for that, and we seem to be opening it in binary mode, so probably it's broken either way. We can resort to `WriteConsoleA()` in curl, but there is a chance this will cut multibyte chars into pieces. `fwrite()` seems to be getting this right, but hard to say without tests.